ruby/test_driver.rb

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env ruby
+
+$LOAD_PATH << '.'
+
+require 'json'
+require 'ionoscloud'
+
+config = Ionoscloud::Configuration.new
+
+config.username = ENV['IONOS_USERNAME']
+config.password = ENV['IONOS_PASSWORD']
+
+api_client = Ionoscloud::ApiClient.new(config)
+
+def underscore_string(str)
+  str.gsub(/::/, '/')
+     .gsub(/([A-Z]+)([A-Z][a-z])/, '\1_\2')
+     .gsub(/([a-z\d])([A-Z])/, '\1_\2')
+     .tr('-', '_')
+     .downcase
+end
+
+def get_class(operation)
+  Ionoscloud.constants.select do |c|
+    Ionoscloud.const_get(c).is_a? Class and Ionoscloud.const_get(c).name.end_with? 'Api'
+  end.each do |class_symbol|
+    class_name = class_symbol.to_s
+    methods = Ionoscloud.const_get(class_name).new.methods - Object.methods
+    return Ionoscloud.const_get(class_name) if methods.map(&:to_s).include? operation
+  end
+  nil
+end
+
+input = gets.chomp
+testing_data = JSON.parse(input.gsub('\"', '"'))
+
+operation = testing_data['operation']
+
+if operation == 'waitForRequest'
+  request_id = testing_data['params'][0]['value'].scan(%r{/requests/(\b[0-9a-f]{8}\b-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-\b[0-9a-f]{12}\b)}).last.first
+
+  api_client.wait_for_completion(request_id)
```

Comment:
what happens if `waitForCompletion` throws an error? we should report the error to stdout in the `error` node of the json.

Reply:
Done. Also changed it for the python driver.